src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -100,16 +100,21 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
 		if (query != null) {
-			boolQB.should(QueryBuilders.matchQuery("name", query).boost(4.0f));
-			boolQB.should(QueryBuilders.prefixQuery("_all", query).boost(1.5f));
-			boolQB.should(QueryBuilders.matchPhrasePrefixQuery("_all", query).slop(2).fuzziness("AUTO"));
+			//Following clause return a module whose UID matches exactly
+			boolQB.should(QueryBuilders.termQuery("_id", query).boost(1500.0f));
+			boolQB.should(QueryBuilders.prefixQuery("name", query).boost(4.0f));
+			boolQB.should(QueryBuilders.termQuery("tags", query).boost(1500.0f));
```

Comment:
Also, sort these by boost, so that someone reading the code can immediately see the most important clauses first.